torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -157,6 +157,15 @@ std::shared_ptr<NCCLComm> NCCLComm::split(
       "Color must be a non-negative value or NCCL_SPLIT_NOCOLOR (-1)"
       ", but got ",
       color_id);
+#ifdef NCCL_COMM_SPLIT_GROUP_RANKS_SUPPORTED
+  // pass group_ranks info to nccl through nccl_config
+  config.splitGroupSize = ranks_ull.size();
+  config.splitGroupRanks = new int[ranks_ull.size()];
```

Comment:
Dynamic Arrays with new placement, yikes. Any better way to transfer that buffer?